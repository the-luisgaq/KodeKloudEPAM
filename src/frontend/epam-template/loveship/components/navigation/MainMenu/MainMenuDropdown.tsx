import { createSkinComponent } from '@epam/uui-core';
import * as uui from '@epam/uui';
import type { MainMenuDropdownProps as Props } from '@epam/uui-components';

export interface MainMenuDropdownProps extends Props {
    /**
     * Defines component color scheme.
     * @default 'dark'
     */
    color?: 'white' | 'dark';
}

function applyMainMenuMods(mods: MainMenuDropdownProps) {
    return [
        mods.color && `uui-main_menu-${mods.color || 'dark'}`,
    ];
}

export const MainMenuDropdown = createSkinComponent<Props, MainMenuDropdownProps>(
    uui.MainMenuDropdown,
    undefined,
    applyMainMenuMods,
);
