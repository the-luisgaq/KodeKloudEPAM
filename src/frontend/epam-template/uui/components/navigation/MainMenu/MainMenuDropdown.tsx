import * as React from 'react';
import FocusLock from 'react-focus-lock';
import cx from 'classnames';
import { Dropdown, MainMenuDropdownProps } from '@epam/uui-components';
import { MainMenuButton } from './MainMenuButton';
import css from './MainMenuDropdown.module.scss';

export class MainMenuDropdown extends React.Component<MainMenuDropdownProps> {
    render() {
        return (
            <Dropdown
                renderTarget={ (props) => (
                    <MainMenuButton
                        caption={ this.props.caption }
                        { ...props }
                        rawProps={ this.props.rawProps }
                        isLinkActive={ this.props.isLinkActive }
                        isDropdown
                    />
                ) }
                renderBody={ (props) => {
                    const handleEscape = (e: React.KeyboardEvent<HTMLElement>) => {
                        if (e.key === 'Escape' && props.isOpen) {
                            props.onClose();
                        }
                    };

                    return (
                        <FocusLock returnFocus persistentFocus lockProps={ { onKeyDown: handleEscape } }>
                            <div className={ cx(css.dropdownBody, 'uui-main_menu-dropdown', this.props.cx) }>
                                {this.props.renderBody
                                    ? this.props.renderBody({ ...props })
                                    : React.Children.map<React.ReactElement<any>, React.ReactElement<any>>(this.props.children as React.ReactElement<any>[], (item) => {
                                        if (!item) return item;
                                        return React.createElement(item.type, {
                                            ...item.props,
                                            onClick: item.props.onClick
                                                ? () => {
                                                    item.props.onClick();
                                                    props.onClose();
                                                }
                                                : null,
                                        });
                                    })}
                            </div>
                        </FocusLock>
                    );
                } }
                placement="bottom-start"
            />
        );
    }
}
