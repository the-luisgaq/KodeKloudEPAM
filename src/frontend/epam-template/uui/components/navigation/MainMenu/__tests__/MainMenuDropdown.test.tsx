import React from 'react';
import { renderSnapshotWithContextAsync } from '@epam/uui-test-utils';
import { MainMenuDropdown } from '../MainMenuDropdown';
import { MainMenuButton } from '../MainMenuButton';

describe('MainMenuDropdown', () => {
    it('should be rendered correctly with minimum props', async () => {
        const tree = await renderSnapshotWithContextAsync(<MainMenuDropdown />);
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', async () => {
        const tree = await renderSnapshotWithContextAsync(
            <MainMenuDropdown caption="Test button" estimatedWidth={ 120 } priority={ 6 }>
                <MainMenuButton collapseToMore caption="Impact" />
                <MainMenuButton collapseToMore caption="ENGX" />
                <MainMenuButton collapseToMore caption="Cloud" />
            </MainMenuDropdown>,
        );
        expect(tree).toMatchSnapshot();
    });
});
