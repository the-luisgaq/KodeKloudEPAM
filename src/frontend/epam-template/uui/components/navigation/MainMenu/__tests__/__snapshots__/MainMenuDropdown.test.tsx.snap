// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuDropdown should be rendered correctly 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    class="uui-button-box uui-enabled -clickable root type-primary"
    tabindex="0"
    type="button"
  >
    <div
      class="uui-caption"
    >
      Test button
    </div>
    <div
      class="container uui-icon uui-enabled"
    >
      <svg
        class=""
      />
    </div>
  </button>
</DocumentFragment>
`;

exports[`MainMenuDropdown should be rendered correctly with minimum props 1`] = `
<DocumentFragment>
  <button
    aria-expanded="false"
    class="uui-button-box uui-enabled -clickable root type-primary"
    tabindex="0"
    type="button"
  >
    <div
      class="container uui-icon uui-enabled"
    >
      <svg
        class=""
      />
    </div>
  </button>
</DocumentFragment>
`;
